edx_proctoring/signals.py

```diff
@@ -0,0 +1,124 @@
+"edx-proctoring signals"
+from django.db.models.signals import pre_save, post_save, pre_delete
+from django.dispatch import receiver
+from edx_proctoring import models
+from edx_proctoring.backends import get_backend_provider
+
+
+@receiver(pre_save, sender=models.ProctoredExam)
+def check_for_category_switch(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    If the exam switches from proctored to timed, notify the backend
+    """
+    if instance.id:
+        original = sender.objects.get(pk=instance.id)
+        if original.is_proctored and instance.is_proctored != original.is_proctored:
+            from edx_proctoring.serializers import ProctoredExamSerializer
+            exam = ProctoredExamSerializer(instance).data
+            exam['is_active'] = False
+            exam['is_proctored'] = True
+            # we have to pretend that the exam is still proctored
+            # or else we get_backend_provider will return None
```

Comment:
s/we //

Reply:
What's unclear from the code even with this comment: the is_proctored field being set here won't affect what backend.on_exam_saved does

I think this would be cleared up by eliminating the need for pretending by adding a get_backend_provider_by_name (or the like) to pull out the final line of get_backend_provider. That way there's no room to worry about whether the backend will care about whether is_proctored is set.

Reply:
I changed get_backend_provider to return a backend for the given name. Good idea.